src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -3286,13 +3285,33 @@ public bool PackageExistsInPackagesFolder(PackageIdentity packageIdentity)
             return new ResolvedPackage(latestVersion, resolvedPackages.Any(p => p.Exists));
         }
 
+        ///A function that given a list of packages, a version and resolution context returns a list
+        /// of packages containing only the packages that follow the required version by the constraints
+        private static List<SourcePackageDependencyInfo> FilterPackagesByVersion(
+            List<SourcePackageDependencyInfo> packages,
+            NuGetVersion version,
+            ResolutionContext resolutionContext)
+        {
+            var result = new List<SourcePackageDependencyInfo>();
```

Comment:
you can do this whole thing using lync:

var result = packages.Where(p => PrunePackageTree.MeetsVersionConstraints(p.Version, version, resolutionContext.VersionConstraints))